src/lib.rs

```diff
@@ -343,7 +343,8 @@ impl<T: Primitive> AtomicMaybeUninit<T> {
     }
 
     /// Stores a value into the atomic integer if the current value is the same as
```

Comment:
FWIW I found it a bit confusing that all the functions talk about "atomic integer" even though they are generic over `T`.

Reply:
Good catch. Changed to "atomic value" in 14f49177f3fde9c1c8be05b005f17a07d4960aee (`new` and `from` were already using it).